std/machines/hash/poseidon2_gl.asm

```diff
@@ -143,16 +157,25 @@ machine Poseidon2GL(mem: Memory) with
         input,
     );
 
-    // Write the output to memory at the next time step
-    let output_addr;
+    // Decide which halves to output:
+    let output_halves;
+    let output_first_half = new_bool();
+    let output_second_half = new_bool();
+    output_halves = output_first_half + 2 * output_second_half;
```

Comment:
Do we need support for both and none? I think it's only used in the test?

If we only supported exactly one half, we could:
- Receive `output_second_half` directly as input (saves a column)
- Define `let output_first_half = 1 - output_second_half` (saves another column)
- Remove 4 links to memory

Reply:
I tried doing this way, but I couldn't because the index of `output[index]` must be constant, so all 8 links are needed regardless...

Now thinking better, I could create a new array[4], and assign the values depending on the `output_first_half`, but then the column balance would be +3.

Would it be worth?

Besides, the original form of sponge construction (not the one used in plonky3) requires the full state. So there is at least one theoretical use.

Reply:
I think this would work:
```
let output_first_half = 1 - output_second_half;
let output_0 = output_first_half * output[0] + output_second_half * output[4];
link if is_used ~> mem.mstore(output_addr + 0, output_time_step, output_0);
```
which would not introduce any extra columns, right?

Reply:
Regarding whether we'll need the full state, I don't know :)

Reply:
@georgwiese Can we merge this as is?